lib/matplotlib/image.py

```diff
@@ -419,14 +428,18 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 # colormapping works correctly
                 hid_output = output
                 output = np.ma.masked_array(
-                    hid_output[..., 0], hid_output[..., 3] < 0.5)
+                    hid_output[..., 0], hid_output[..., 3] == 0)
```

Comment:
Yes, this line is indeed causing the problem. In particular, the alpha channel is interpolated near the boundary of an image. The `< 0.5` implementation will set pixels that are "more than half transparent" to fully transparent whereas the `== 0` implementation only treats them as masked if they are fully transparent.

In short, the boundaries in a few tests differ slightly.

![rotate_image-failed-diff](https://cloud.githubusercontent.com/assets/966348/23555782/1fce8a54-0022-11e7-920a-69363d219ed0.png)

I believe the `== 0` implementation is preferable because it effectively provides anti aliasing on the boundary by restoring the alpha channel. Here is a zoomed-in example.

![screen shot 2017-03-03 at 15 03 01](https://cloud.githubusercontent.com/assets/966348/23555913/8c53c2ac-0022-11e7-8d9f-d2e77bcb6864.png)
![screen shot 2017-03-03 at 15 03 06](https://cloud.githubusercontent.com/assets/966348/23555914/8c557bce-0022-11e7-8715-0e7d12553493.png)